warehouse/macaroons/caveats/__init__.py

```diff
@@ -149,6 +145,104 @@ def verify(self, request: Request, context: Any, permission: str) -> Result:
         return Success()
 
 
+class PublicPermissions(enum.IntFlag, boundary=enum.FlagBoundary.STRICT):
+    # We use 1 << N here instead of enum.auto() because the selected value is
+    # very important to us, if the value of a permission changes, then the scope
+    # of existing Macaroons will change, which would be a major security fail.
+    #
+    # Explicitly setting our values also means that we can skip values, allowing
+    # us to reserve earlier bits for permissions that are likely going to be
+    # widely used, and push less commonly used permissions to later bits.
+    #
+    # The structure of these values should basically always be 1 << N, where N
+    # is a unique integer per permission, starting with 0 and increasing from
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought (non-blocking):** while the list is currently small, should we add the [`@unique` decorator](https://docs.python.org/3/library/enum.html#enum.unique) to prevent accidental duplicate values? (self-note - probably a good idea to add to the `Permissions` enum too!)